server/aws-lsp-codewhisperer/src/language-server/telemetryService.ts

```diff
@@ -0,0 +1,83 @@
+import { CodeWhispererServiceToken } from './codeWhispererService'
+import { CredentialsProvider, Telemetry } from '@aws/language-server-runtimes/server-interface'
+import { CodeWhispererSession } from './session/sessionManager'
+import {
+    SuggestionState,
+    UserTriggerDecisionEvent,
+    UserContext,
+    OptOutPreference,
+} from '../client/token/codewhispererbearertokenclient'
+import { getCompletionType } from './utils'
+
+export class TelemetryService extends CodeWhispererServiceToken {
+    private userContext!: UserContext
+    private optOutPreference!: OptOutPreference
+    private enableTelemetryEventsToDestination!: boolean
+    private toolkitTelemetry!: Telemetry
+
+    constructor(credentialsProvider: CredentialsProvider, additionalAwsConfig: AWS.ConfigurationOptions = {}) {
+        super(credentialsProvider, additionalAwsConfig)
+    }
+
+    public setToolkitTelemetry(toolkitTelemetry: Telemetry): void {
+        this.toolkitTelemetry = toolkitTelemetry
```

Comment:
I do not think we should name it as a `**toolkit**Telemetry`.

Reply:
clientTelemetry?

Reply:
discussed this with Viktor to pass just as telemetry and in the constructor